### ISSUE TITLE
"Configure-EdgeExtension -configureMode add" Error

### ISSUE THREAD

**Issue by Tirakk** [PSAppDeployToolkit/PSAppDeployToolkit#931]
**Describe the bug**
I tried the new function "Configure-EdgeExtension", but it throws an error from the AppDeployToolkitMain.ps1 Line 16524
On my system the "HKLM:\SOFTWARE\Policies\Microsoft\Edge -> ExtensionSettings" Value wasnt existent and was created by the toolkit, but with empty data.
The following variable $installedExtensions is then empty and "Add-Member" in line 16524 throws the following error:

![image](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/77411182/9b033fc3-3890-4669-8673-85ad9eb6d0fd)

I fixed it on my side by writing the default value to "ExtensionSettings" and removing the following Else{} to update the $installedExtensions Variable.

**To Reproduce**
1. Removing "ExtensionSettings" if the value exists
2. Trying to add a EdgeExtension with "Configure-Edgeextension -ExtensionID "extensionID" -InstallationMode "force_install" -configuremode 'add' -UpdateUrl "https://www.contoso.com/extension"

**Screenshots**

**Toolkit Version:**
3.10.0

**Powershell Version:**
5.1

**Comment by xstaindx** [comment]
Just joining this thread from @seanlillis @sintaxasn per twitter. https://twitter.com/seanels/status/1775835974917816720

I am attemtping to install ad-block plus or adobe pdf either one, does not show up in Edge.   I am using WIndows 10 Pro 22H2.

Here is the code I used: (.ps1 in zip)

Configure-EdgeExtension -configureMode Add -extensionID gmgoamodcdcjnbaobigkjelfplakmdhh -InstallationMode force_installed -UpdateUrl https://edge.microsoft.com/extensionwebstorebase/v1/crx

Log File (in zip)
Output photos (in zip)

[AdBlockPlusAttempt.zip](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/files/14865583/AdBlockPlusAttempt.zip)
![GKUR2f6aAAAc2rJ](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/25702100/b942946b-8a65-4462-b8e4-0d68dbeeedcd)
![GKURx9Ka0AAfgIN](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/25702100/aea114cb-ac44-4a3d-a610-9734acc4e377)
![GKUQ6t_aYAAtgia](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/25702100/dc7d581c-f706-4a5e-912c-e8905712bddf)
[PS_AppDeployToolkitMain_3.10.0_EN_01_PSAppDeployToolkit_Install.log](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/files/14865586/PS_AppDeployToolkitMain_3.10.0_EN_01_PSAppDeployToolkit_Install.log)

**Comment by Tirakk** [comment]
> Just joining this thread from @seanlillis @sintaxasn per twitter. https://twitter.com/seanels/status/1775835974917816720
> 
> 
> 
> I am attemtping to install ad-block plus or adobe pdf either one, does not show up in Edge.   I am using WIndows 10 Pro 22H2.
> 
> 
> 
> Here is the code I used: (deploy-application.ps1 in zip)
> 
> 
> 
> Configure-EdgeExtension -configureMode Add -extensionID gmgoamodcdcjnbaobigkjelfplakmdhh -InstallationMode force_installed -UpdateUrl https://edge.microsoft.com/extensionwebstorebase/v1/crx
> 
> 
> 
> Output photos (in zip)
> 
> 
> 
> [AdBlockPlusAttempt.zip](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/files/14865583/AdBlockPlusAttempt.zip)
> 
> ![GKUR2f6aAAAc2rJ](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/25702100/b942946b-8a65-4462-b8e4-0d68dbeeedcd)
> 
> ![GKURx9Ka0AAfgIN](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/25702100/aea114cb-ac44-4a3d-a610-9734acc4e377)
> 
> ![GKUQ6t_aYAAtgia](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/assets/25702100/dc7d581c-f706-4a5e-912c-e8905712bddf)
> 
> [PS_AppDeployToolkitMain_3.10.0_EN_01_PSAppDeployToolkit_Install.log]
> 
> 
> 
> LOG FILE:
> 
> 
> 
> (https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/files/14865586/PS_AppDeployToolkitMain_3.10.0_EN_01_PSAppDeployToolkit_Install.log)

Yeah kinda related - in your log you can see that you ExtensionSettings value also gets written with empty data

**Comment by seanlillis** [comment]
This is now fixed in the develop branch and will be included in the next release. I've had to change some parameter names to facilitate the addition of Parameter Sets to ensure correct grouping of mandatory and non-mandatory parameters for Add/Remove.
